src/Tools/NodeDocumentationMarkdownGenerator/Commands/RenameCommand.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.IO;
+using System.Linq;
+using NodeDocumentationMarkdownGenerator.Verbs;
+
+namespace NodeDocumentationMarkdownGenerator.Commands
+{
+    internal static class RenameCommand
+    {
+        internal static void HandleRename(RenameOptions opts)
+        {
+            if (opts.InputMdFile is null && opts.InputMdDirectory != null)
+            {
+                RenameDirectory(opts.InputMdDirectory, opts.MaxLength);
+            }
+            else if (opts.InputMdFile != null && opts.InputMdDirectory is null)
+            {
+                RenameFile(opts.InputMdFile);
+            }
+            else
+            {
+                Console.WriteLine("Invalid options: You can rename a single file using the file option\nor renaming multiple files in a directory (if they are longer than max length)\nusing the directory option");
+            }
+        }
+
+        private static void RenameFile(string file)
+        {
+            var extension = Path.GetExtension(file);
+            if (!extension.Equals(".md", StringComparison.InvariantCultureIgnoreCase))
+            {
+                Console.WriteLine($"Can only rename MD files: {file}");
+                return;
+            }
+
+            if (!File.Exists(file))
+            {
+                Console.WriteLine($"File not found: {file}");
+                return;
+            }
+
+            var baseName = Path.GetFileNameWithoutExtension(file);
+            var shortName = Dynamo.Utilities.Hash.GetHashFilenameFromString(baseName);
+
+            RenameFile(file, baseName, shortName);
+        }
+
+        private static void RenameFile(string file, string baseName, string shortName)
+        {
+            var content = File.ReadAllText(file);
+            content = content.Replace(baseName, shortName);
```

Comment:
Are you saving the hashed name in the file contents as well?

Reply:
Nop, only the original filename. But I could easily do it.

Reply:
I meant to ask what is the purpose of line 50 if it's not necessary to save the hashed name in the file?